probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+
+        Args :
+            clf(model object):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategies (dictionary of sklearn.impute objects):
+                Dictionary containing the sklearn.impute objects.
+                e.g.
+                
+                strategies = {'KNN' : KNNImputer(n_neighbors=3),
+                'Simple Median Imputer' : SimpleImputer(strategy='median',add_indicator=True),
+                'Iterative Imputer'  : IterativeImputer(add_indicator=True,n_nearest_features=5,
+                sample_posterior=True)}
+                This allows you to have fine grained control over the imputation method.
+
+            scoring (string, list of strings, probatus.utils.Scorer or list of probatus.utils.Scorers, optional):
+                Metrics for which the score is calculated. It can be either a name or list of names metric names and
+                needs to be aligned with predefined [classification scorers names in sklearn](https://scikit-learn.org/stable/modules/model_evaluation.html).
+                Another option is using probatus.utils.Scorer to define a custom metric.
+
+            model_na_support(boolean): default True
+                If the classifier supports missing values by default e.g. LightGBM,XGBoost etc. 
+                If True an default comparison `Model Imputation` will be added indicating the results without any explict imputation. 
+                If False only the provided strategies will be used.
+            
+            n_jobs (int, optional):
+                Number of cores to run in parallel while fitting across folds. None means 1 unless in a
+                `joblib.parallel_backend` context. -1 means using all processors.
+
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - nether prints nor warnings are shown
+                - 1 - 50 - only most important warnings regarding data properties are shown (excluding SHAP warnings)
+                - 51 - 100 - shows most important warnings, prints of the feature removal process
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+            
+            random_state (int, optional):
+                Random state set at each round of feature elimination. If it is None, the results will not be
+                reproducible and in random search at each iteration a different hyperparameters might be tested. For
+                reproducible results set it to integer.
+        """
+        self.clf = clf
+        self.model_na_support = model_na_support
+        self.scorer = get_single_scorer(scoring)
+        self.strategies = strategies
+        if cv is None:
+            self.cv = 5
+        else :
+            self.cv = cv
+        self.verbose = verbose
+
+        self.n_jobs = n_jobs
+        
+        if random_state is None:
```

Comment:
Please check how is the random state implemented in other modules. I guess if it is None, then we should not set it, because every run should be different then.

Reply:
Done.